Sources/SwiftSyntax/generated/SyntaxTraits.swift

```diff
@@ -16,11 +16,17 @@
 
 
 public protocol BracedSyntax: SyntaxProtocol {
+  /// ### Tokens
+  /// 
+  /// For syntax trees generated by the parser, this is guaranteed to be one of the following kind: `'{'`
```

Comment:
Not sure if I think the `### Tokens` is needed? 
@ahoppen what is the reason to use this?

Reply:
My gut feel is there are two reasons for it:
- It's consistent with how we structure documentation for `Node`
- Using these headings works nice in DocC when it's rendered.

I, personally, would like to keep them, but I see how it might feel like an overkill for a situation where there's one possible token kind.

Reply:
Nice with DocC. 
I will need to see how it renders then 😁

Reply:
So in `SyntaxNodesFile.swift` we construct node documentation of three pieces. Here's the `node.grammaer` piece that generates the Children subheading pictured below: https://github.com/apple/swift-syntax/blob/ddd05efdf15574413c70645803488f2c4311014c/CodeGeneration/Sources/SyntaxSupport/Node.swift#L357

<img width="1612" alt="Screenshot 2023-09-12 at 11 57 02 AM" src="https://github.com/apple/swift-syntax/assets/43633/e4efbafc-e14c-4fbb-90df-3b5cd42c97bb">

Ahem... should we put them in a table? Should we also backtich the children names? Oh gawd why have I looked at it.

Reply:
I just saw the children. Looks nice. 

But I think they are a bit duplicate of "Instance Properties". 
But that is not related to this PR

Reply:
One of the follow-up items on this PR is to make _how_ we render those pieces (children) in the code consistent between Node and Children. I might do that in this PR, or in another one. 

Since I'm already elbow deep in this, I'm happy to help to explore how we can improve the autogenerated docs, here and otherwise. Feel free to write / assign issues to me.

Reply:
> But I think they are a bit duplicate of "Instance Properties".
> But that is not related to this PR

It kind of is but the *Children* section has a few properties that the *Instance Properties* section doesn’t have
- It’s in the order that the children appears within the syntax node, compared to being alphabetical, which is a huge point
- It shows the token kinds for all token children (which might be less important with this PR) but it’s nice to see that in a concise way.
- It omits the `unexpected` children, which is kind of nice.

Reply:
When I search for documentation I would actually argue it's easier for users to find it when ordered alphabetical, than in the order they appear in the code, as I don't see the code right next to the docs (I use SPI). 
We know the order, as we look into the code a lot. But I'm not sure that people are thinking on that order in the code, when reading the docs. 

To point two we could add that to the instance properties. 
Auto generate that and add it as part of the provided documentation? 

The `unexpected` I agree that's nice, for me it's actually a bigger point that it's omitted that they ordered alphabetical.

Reply:
> When I search for documentation I would actually argue it's easier for users to find it when ordered alphabetical, than in the order they appear in the code, as I don't see the code right next to the docs (I use SPI).
> We know the order, as we look into the code a lot. But I'm not sure that people are thinking on that order in the code, when reading the docs.

At least for me, it makes a huge difference to have them ordered in the same order that the children will occur in the parsed source code. It might be a silly example but if I’m looking at `FunctionDeclSyntax` and want to know which node represents `public`, I know that `public` is spelled before `func`. Thus, I look for the child that represents `func` and then just look at the nodes before it: `attributes` and `modifiers`. I know that attributes are spelled with an `@`, so I know that `modifiers` need to contain `public`. And I didn’t need to wonder if `public` might be part of `signature`.

> To point two we could add that to the instance properties. And if there is added some documentation inside the Node or Child in SyntaxSupport, I would just append it to the end. So the handwritten is first, and then the generated documentation.

We could, but at least with the current rendering of docc, I would argue that the `Children` section is denser and more readable. Also, just for some background, the *Children* section got introduced as a replacement for the grammar comments that we had scattered throughout the source code and I think it does fulfill that role.